packages/client/src/containers/apiTokenPage.tsx

```diff
@@ -14,173 +26,182 @@ const breadcrumbs: BreadcrumbItemSetup[] = [
     title: 'API token',
     tips: 'A valid token is mandatory to allow 3rd-party using PrimeHub APIs.',
     tipsLink: 'https://docs.primehub.io/docs/tasks/api-token'
-  }
+  },
 ];
 
-type Props = {
-  client: ApolloClient<any>;
+interface Props {
+  downloadConfig?: (config) => void;
 }
 
-type State = {
-}
-
-const GET_API_TOKEN_COUNT = gql`
-  query getApiTokenCount {
-    me {
-      apiTokenCount
-    }
-  }
-`;
-
-const REVOKE_API_TOKEN = gql`
-  mutation revokeApiToken {
-    revokeApiToken {
-      id
-    }
-  }
-`;
-
-class ApiTokenPage extends React.Component<Props, State> {
-  state = {
-    modal: false
-  }
+export default function ApiTokenPage(props: Props) {
+  const graphqlEndpoint = window.absGraphqlEndpoint;
+  const apiToken = window.apiToken;
+  const example = `\nAPI_TOKEN="${apiToken ? apiToken : '<API TOKEN>'}"
 
-  apiToken: any;
-  graphqlEndpoint: string;
-  requestApiTokenEndpoint: string;
-
-  refExample: React.RefObject<any> = React.createRef();
-  refToken: React.RefObject<any> = React.createRef();
-
-  constructor(props) {
-    super(props)
-
-    this.graphqlEndpoint = window.absGraphqlEndpoint;
-    this.apiToken = window.apiToken;
-    this.requestApiTokenEndpoint = window.requestApiTokenEndpoint;
-  }
-
-  handleRequestApiToken = () => {
-    const {client} = this.props;
-    client.query<any>({query: GET_API_TOKEN_COUNT, fetchPolicy: 'no-cache'})
-    .then((result) => {
-      if (result.data.me.apiTokenCount == 0) {
+curl -X POST \\
+  -H 'Content-Type: application/json' \\
+  -H "authorization: Bearer \${API_TOKEN}" \\
+  -d '{"query":"{me{id,username}}"}' \\
+  ${graphqlEndpoint}`;
+
+  const [statusCopyToken, copyToken] = useClipboard({
+    text: apiToken,
+    timeout: 2000,
+  });
+  const [, copyExample] = useClipboard({
+    text: example,
+    timeout: 2000,
+  });
+
+  const handleRequestApiToken = async (client: ApolloClient<any>) => {
+    const requestApiTokenEndpoint = window.requestApiTokenEndpoint;
+
+    const requestToken = async () => {
```

Comment:
Just advice, you can write logic into line 72, so you don't need to declare the function.

Reply:
This function is used twice. at #72 and #80